src/Export/VTK.jl

```diff
@@ -1,3 +1,135 @@
+
+"""
+    VTKFile(filename::AbstractString, grid::AbstractGrid; kwargs...)
+    VTKFile(filename::AbstractString, dh::DofHandler; kwargs...)
+
+Create a `VTKFile` that contains an unstructured VTK grid. 
+The keyword arguments are forwarded to `WriteVTK.vtk_grid`, see 
+[Data Formatting Options](https://juliavtk.github.io/WriteVTK.jl/stable/grids/syntax/#Data-formatting-options)
+
+This file handler can be used to to write data with
+
+* [`write_solution`](@ref)
+* [`write_celldata`](@ref)
+* [`write_projection`](@ref)
+* [`write_nodedata`](@ref).
+* [`Ferrite.write_cellset`](@ref)
+* [`Ferrite.write_nodeset`](@ref)
+* [`Ferrite.write_dirichlet`](@ref)
+
+It is necessary to call `close(::VTKFile)` to save the data after writing 
+to the file handler. Using the supported `do`-block does this automatically:
+```julia
+VTKFile(filename, grid) do vtk
+    write_solution(vtk, dh, u)
+    write_celldata(vtk, celldata)
+end
+```
+"""
+struct VTKFile{VTK<:WriteVTK.VTKFile}
+    vtk::VTK
+end
+function VTKFile(filename::String, dh::DofHandler; kwargs...)
+    return VTKFile(filename, get_grid(dh); kwargs...)
+end
+function VTKFile(filename::String, grid::AbstractGrid; kwargs...)
+    vtk = create_vtk_grid(filename, grid; kwargs...)
+    return VTKFile(vtk)
+end
+# Makes it possible to use the `do`-block syntax
+function VTKFile(f::Function, args...; kwargs...)
+    vtk = VTKFile(args...; kwargs...)
+    try
+        f(vtk)
+    finally
+        close(vtk)
+    end
+end
+
+Base.close(vtk::VTKFile) = WriteVTK.vtk_save(vtk.vtk)
```

Comment:
Perhaps this should be added as a finalizer to the `VTKFile` struct? Or is it already a finalizer added for the upstream struct?

Reply:
Do you mean https://docs.julialang.org/en/v1/base/base/#Base.finalizer?

I cannot get this to work for a `struct` (complains not mutable), and I'm also not sure exactly what it would do. I don't know exactly how such finalizers works, but I thought that the following sufficed? 
https://github.com/Ferrite-FEM/Ferrite.jl/blob/692cf786be4e6d6463f67a5ec97c1687eb5ce3cd/src/Export/VTK.jl#L40-L47

(But no hits when searching for `finalizer` in the WriteVTK.jl repo, so probably not in the upstream struct)

Reply:
> struct (complains not mutable),

Yea, you would have to make it `mutable struct`.

> I'm also not sure exactly what it would do.

Call `vtk_save` (or our `close` wrapper)

> I don't know exactly how such finalizers works, but I thought that the following sufficed?

Yea if you use the `do`-block, but some examples calls `close` directly for example. This could be a separate PR anyway, let's think on it.